src/Microsoft.VisualStudio.Web.CodeGenerators.Mvc/Controller/ControllerGeneratorBase.cs

```diff
@@ -83,13 +84,31 @@ protected virtual IEnumerable<string> TemplateFolders
             }
         }
 
-        protected string GetControllerNamespace()
+        protected string GetDefaultControllerNamespace()
         {
             // Review: MVC scaffolding used ActiveProject's MSBuild RootNamespace property
             // That's not possible in command line scaffolding - the closest we can get is
             // the name of assembly??
             var appName = ApplicationInfo.ApplicationName;
-            return appName + "." + Constants.ControllersFolderName;
+
+            var defaultControllerNamespace = appName + "." + Constants.ControllersFolderName;
```

Comment:
Its a workaround for both, but since we have access to the relative path, we can try to use it to determine the default namespace. 

---
In reply to: [64578230](https://github.com/aspnet/Scaffolding/pull/238#discussion_r64578230) [](ancestors = 64578230)